extension/src/setup/index.ts

```diff
@@ -160,6 +151,12 @@ export class Setup
     this.setCommandsAvailability(false)
     this.setProjectAvailability()
 
+    this.studio = new Studio(
+      internalCommands,
+      this.studioConnectionChanged,
```

Comment:
[F] Unless you are planning to use `this.studioConnectionChanged` to fire events outside of the Studio class I would create the event emitter insider of the class and wire up the listener with 
```
this.onDidChangeStudioConnection = this.studio.onDidChangeStudioConnection
```

Reply:
Good idea!